#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: try to keep the stack word aligned

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

---
 gcc/config/m68k/m68k.h |    1 +
 1 file changed, 1 insertion(+)

Index: gcc-4.2/gcc/config/m68k/m68k.h
===================================================================
--- gcc-4.2.orig/gcc/config/m68k/m68k.h
+++ gcc-4.2/gcc/config/m68k/m68k.h
@@ -148,6 +148,7 @@ along with GCC; see the file COPYING3.  
 
 #define PARM_BOUNDARY (TARGET_SHORT ? 16 : 32)
 #define STACK_BOUNDARY 16
+#define PREFERRED_STACK_BOUNDARY 32
 #define FUNCTION_BOUNDARY 16
 #define EMPTY_FIELD_BOUNDARY 16
 
